src/utility/StringAdvanceMethod.java

```diff
@@ -1,3 +1,50 @@
 package utility;
 
 // Type your code
+public class StringAdvanceMethod{
+	private String input1;
+	private String input2;
+	
+	public String getInput1() {
+		return input1;
+	}
+	public void setInput1(String input1) {
+		this.input1 = input1;
+	}
+	public String getInput2() {
+		return input2;
+	}
+	public void setInput2(String input2) {
+		this.input2 = input2;
+	}
+	
+	public String concat(String input1,String input2)
+	{
+		if(input1==null||input2==null)
+		{
+			return "";
+		}
+		return input1+input2;
+	}
+	
+	public String split(String input1,String seperator)
+	{
+		String seperatedString=String.join("  ",input1.split(seperator));
+		return seperatedString;
+	}
+	
+	public String indexOf(String input1,String input2)
+	{
+		if(input1==null||input2==null||input1==""||input2=="")
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.

Reply:
Got it. Thank you, will change it.